Dockerfile

```diff
@@ -1,22 +1,22 @@
-FROM ruby:2.4.2
+FROM ruby:2.4.10
 
-# Default node version on apt is old. This makes sure a recent version is installed
-# This step also runs apt-get update
+# Install Node and other build dependencies
 RUN curl -sL https://deb.nodesource.com/setup_12.x | bash -
-RUN apt-get install -y --force-yes build-essential libpq-dev nodejs
+RUN apt-get install -y build-essential libpq-dev nodejs
 
-# Install latest chrome dev package
-RUN set -ex; \
-    wget -q -O - https://dl.google.com/linux/linux_signing_key.pub | apt-key add - \
-    && wget -q -O - https://deb.nodesource.com/gpgkey/nodesource.gpg.key | apt-key add - \
-    && sh -c 'echo "deb [arch=amd64] http://dl.google.com/linux/chrome/deb/ stable main" >> /etc/apt/sources.list.d/google.list' \
-    && apt-get update \
-    && apt-get install -y --force-yes google-chrome-stable --no-install-recommends \
+# Install latest Chrome browser (for automated testing)
+RUN curl -L https://dl-ssl.google.com/linux/linux_signing_key.pub | apt-key add - \ 
+    && echo "deb http://dl.google.com/linux/chrome/deb/ stable main" >> /etc/apt/sources.list.d/google.list \
     && rm -rf /var/lib/apt/lists/*
+RUN apt-get update && apt-get -y install google-chrome-stable
 
 WORKDIR /planner
 
+# Install project dependencies using all available CPU cores
 COPY Gemfile Gemfile.lock ./
-RUN bundle install
+RUN bundle install --jobs $(getconf _NPROCESSORS_ONLN) --retry 3
+
+# Setup our database
+RUN bundle exec rake db:create db:migrate db:seed db:test:prepare
```

Comment:
Fly by comment: I wouldn't imagine this would work since within the build environment there is no database

Reply:
Yeah, I’m not the most well-versed in Docker, so I was just giving it a stab, see where I’d go… What you said makes sense now and it explains why it was not working! 😬